src/insights_client/tests/test_motd.py

```diff
@@ -0,0 +1,114 @@
+import os
+import pathlib
+import tempfile
+import unittest
+import unittest.mock
+
+import pytest
+
+import insights_client
+
+
+class MockFS:
+    """Mock filesystem.
+
+    Ensures that the files required for the MOTD manipulation are created
+    in a temporary directory instead of working with actual system files.
+    """
+
+    _dir: tempfile.TemporaryDirectory
+
+    def __init__(self):
+        self._dir = tempfile.TemporaryDirectory(prefix="test-chroot-")
+
+        paths: dict[str, str] = {
+            "insights_client.MOTD_SRC": "/etc/insights-client/insights-client.motd",
+            "insights_client.MOTD_FILE": "/etc/motd.d/insights-client",
+            "insights_client.REGISTERED_FILE": "/etc/insights-client/.registered",
+            "insights_client.UNREGISTERED_FILE": "/etc/insights-client/.unregistered",
+        }
+
+        self.mocks: dict[str, unittest.mock.patch] = {}
+        for label, path in paths.items():
+            mocked_path = f"{self.chroot}{path}"
+            pathlib.Path(mocked_path).parent.mkdir(parents=True, exist_ok=True)
```

Comment:
Although I like the object-oriented [pathlib](https://docs.python.org/3.6/library/pathlib.html?highlight=pathlib#module-pathlib) module, calls like these can easily become quite complex. Importing Path directly would help, but that may not be of your taste and may hinder consistency.

```suggestion
            from pathlib import Path
            Path(mocked_path).parent.mkdir(parents=True, exist_ok=True)
```

Using [os.path.dirname](https://docs.python.org/3.6/library/os.path.html#os.path.dirname) and [os.path.makedirs] may make the call even more concise:

```suggestion
            os.path.makedirs(os.path.dirname(mocked_path), exist_ok=True)
```

or even

```suggestion
            from os.path import dirname, makedirs
            makedirs(dirname(mocked_path), exist_ok=True)
```

if combined. To me it looks elegant.

Reply:
I like the object approach, I find it more descriptive than what you are proposing. Plus I dislike the `((value), arg=value)` it causes, it you have to keep more context in mind because one function is returning and other is ingesting. I know that is how some languages work, but in Python it just confuses me.